addons/account/models/account_move.py

```diff
@@ -3124,6 +3124,8 @@ def create(self, vals_list):
         for vals in vals_list:
             move = self.env['account.move'].browse(vals['move_id'])
             vals.setdefault('company_currency_id', move.company_id.currency_id.id) # important to bypass the ORM limitation where monetary fields are not rounded; more info in the commit message
+            if vals.get('currency_id', False) == vals.get('company_currency_id', False):
```

Comment:
It could be simpler I think:

```suggestion
            if vals.get('currency_id') == vals.get('company_currency_id'):
                vals['currency_id'] = False
```